Shared/Samples/Edit with branch versioning/EditWithBranchVersioningView.swift

```diff
@@ -0,0 +1,258 @@
+            ToolbarItemGroup(placement: .bottomBar) {
+                createVersionButton
+                switchVersionButton
+            }
+        }
+        .errorAlert(presentingError: $error)
+    }
+    
+    /// The button for editing the damage type of the selected feature.
+    private var editDamageTypeButton: some View {
+        Button("Edit", systemImage: "pencil") {
+            isSettingDamageType = true
+        }
+        .imageScale(.large)
+        .labelStyle(.iconOnly)
+        .disabled(model.onDefaultVersion)
+        .confirmationDialog("Damage Type", isPresented: $isSettingDamageType, titleVisibility: .visible) {
+            ForEach(DamageType.allCases, id: \.self) { damageType in
+                Button(damageType.label) {
+                    model.selectedFeature?.setAttributeValue(
+                        damageType.rawValue,
+                        forKey: .damageTypeKey
+                    )
+                    selectedAction = .updateFeature
+                }
+            }
+        } message: {
+            Text("Choose a damage type for the building.")
+        }
+    }
+    
+    /// The button for creating a version.
+    private var createVersionButton: some View {
+        Button("Create") {
+            isCreatingVersion = true
+        }
+        .popover(isPresented: $isCreatingVersion) {
+            CreateVersionParametersView(model: model) { parameters in
+                // Makes a new version with the created parameters.
+                versionParameters = parameters
+                selectedAction = .makeVersion
+            }
+            .presentationDetents([.fraction(0.5)])
+            .frame(idealWidth: 320, idealHeight: 200)
+        }
+    }
+    
+    /// The button for switching the current version.
+    private var switchVersionButton: some View {
+        Button("Switch") {
+            isSwitchingVersion = true
+        }
+        .disabled(model.existingVersionNames.count < 2)
+        .confirmationDialog("Versions", isPresented: $isSwitchingVersion, titleVisibility: .visible) {
+            ForEach(model.existingVersionNames, id: \.self) { versionName in
+                Button(versionName) {
+                    selectedAction = .switchToVersion(version: versionName)
+                }
+            }
+        } message: {
+            Text("Choose a version to switch to.")
+        }
+    }
+}
+
+/// An asynchronous action associated with the sample.
+private enum AsyncAction: Equatable {
+    /// Sets up the sample.
+    case setUp
+    
+    /// Makes a version with the current version parameters.
+    case makeVersion
+    
+    /// Switches to a version with an associated version name.
+    case switchToVersion(version: String)
+    
+    /// Select a feature identified from an associated screen and map point.
+    case selectFeature(screenPoint: CGPoint, mapPoint: Point)
+    
+    /// Updates the selected feature in it's feature table.
+    case updateFeature
```

Comment:
We don't usually separate enum cases with a blank line.

```suggestion
    /// Sets up the sample.
    case setUp
    /// Makes a version with the current version parameters.
    case makeVersion
    /// Switches to a version with an associated version name.
    case switchToVersion(version: String)
    /// Select a feature identified from an associated screen and map point.
    case selectFeature(screenPoint: CGPoint, mapPoint: Point)
    /// Updates the selected feature in it's feature table.
    case updateFeature
```